### PR TITLE
543 import api cli

### LINKED ISSUE #543: Import API should be a command line tool
When importing data from S3 with the import API, the process can take a very long time. Currently import is synchronous, and occurs within a single HTTP request.

To avoid failing on timeouts, and in order to give the user a better experience, ~make this API asynchronous with progress monitoring.~ make it a command line tool that prints out its progress,

### PR DESCRIPTION
**Pull request by johnnyaug**
closes #543

### CODE REVIEW

File: onboard/stats.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func (s *Stats) AddCreated(n int64) {
+	s.reportProgress()
+	atomic.AddInt64(s.AddedOrChanged, n)
+}
+
+func (s *Stats) AddDeleted(n int64) {
+	s.reportProgress()
+	atomic.AddInt64(s.Deleted, n)
```
Comment (johnnyaug) [inline onboard/stats.go:48]: check if need to use load on read

File: onboard/catalog_actions.go [hunk 2]
```diff
@@ -18,9 +18,9 @@ const (
 )
 
 type RepoActions interface {
-	ApplyImport(ctx context.Context, it Iterator, dryRun bool) (*InventoryImportStats, error)
+	ApplyImport(ctx context.Context, it Iterator, stats *Stats, dryRun bool) error
```
Comment (johnnyaug) [inline onboard/catalog_actions.go:21]: separate the callback from the stats.
go back to returning the stats.
the callback should be passed here and called from ApplyImport, perhaps with a dedicated object or perhaps with the stats.

Comment (johnnyaug) [reply onboard/catalog_actions.go:21]: also, instead of callback consider a progress-bar-style interface

File: cmd/lakefs/cmd/import.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+const (
+	DryRunFlagName         = "dry-run"
+	ManifestURLFlagName    = "manifest"
+	ManifestURLFormat      = "s3://example-bucket/inventory/YYYY-MM-DDT00-00Z/manifest.json"
+	ImportCmdNumArgs       = 1
+	ImportSpinnerFrequency = 100 * time.Millisecond
+)
+
+var importCmd = &cobra.Command{
```
Comment (johnnyaug) [inline cmd/lakefs/cmd/import.go:33]: add assert on wrong schema db version

File: cmd/lakefs/cmd/import.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	Use:   "import <repository uri> --manifest <s3 uri to manifest.json>",
+	Short: "Import data from S3 to a lakeFS repository",
+	Long:  "Import from an S3 inventory to lakeFS without copying the data.",
+	Args: lakectl.ValidationChain(
+		lakectl.HasNArgs(ImportCmdNumArgs),
```
Comment (johnnyaug) [inline cmd/lakefs/cmd/import.go:38]: extract somehow
uri verification --> to uri package

File: cmdutils/progress.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+func (p *Progress) SetCurrent(n int64) {
+	atomic.StoreInt64(p.current, n)
+}
+
+func (p *Progress) SetTotal(n int64) {
+	atomic.StoreInt64(p.total, n)
+}
+
+func StartMultiBar(r ProgressReporter) *MultiBar {
```
Comment (johnnyaug) [inline cmdutils/progress.go:75]: separate new & start

Comment (johnnyaug) [reply cmdutils/progress.go:75]: no need for channel, make Stop function to stop ticker

File: cmdutils/progress.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func (b *MultiBar) Finish() {
+	close(b.done)
+}
+
+func (b *MultiBar) Refresh(isCompleted bool) {
+	progress := b.reporter.Progress()
+	for _, p := range progress {
+		total := atomic.LoadInt64(p.total)
```
Comment (johnnyaug) [inline cmdutils/progress.go:101]: use api

File: cmdutils/progress.go [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+		isSpinner := false
+		if total <= 0 {
+			isSpinner = true
+			total = atomic.LoadInt64(p.current)
+		}
+		bar, ok := b.mpbBars[p.label]
+		if !ok {
+			labelDecorator := decor.Name(p.label, decor.WC{W: progressBarNameColumnWidth, C: decor.DidentRight})
```
Comment (johnnyaug) [inline cmdutils/progress.go:109]: extract function

File: block/s3/inventory_iterator.go [hunk 8]
```diff
@@ -2,8 +2,12 @@ package s3
 
 import (
 	"errors"
+	"fmt"
+	"strconv"
 	"time"
 
+	"github.com/treeverse/lakefs/cmdutils"
+
```
Comment (johnnyaug) [inline block/s3/inventory_iterator.go:10]: delete spaces

**Review by nopcoder — APPROVED** [review]

**Pull request merged**